2996_reflection/reflection.md

```diff
@@ -1823,6 +1819,48 @@ Change the grammar for `$operator-or-punctuator$` in paragraph 1 of [lex.operato
 ```
 :::
 
+### [basic.types.general]
+
+Change the first sentence in paragraph 9 of [basic.types.general]{.sref} as follows:
+
+::: bq
+[9]{.pnum} Arithmetic types (6.8.2), enumeration types, pointer types, pointer-to-member types (6.8.4),[ `std::meta::info`,]{.addu} `std::nullptr_t`, and cv-qualified (6.8.5) versions of these types are collectively called scalar types.
+:::
+
+Add a new paragraph at the end of [basic.types.general]{.sref} as follows:
+
+::: bq
+::: addu
+
+[*]{.pnum} A type is *consteval-only* if it is:
+
+  - `std::meta::info`
+  - an array type with a consteval-only element type
+  - a class type with a consteval-only base class type or consteval-only nonstatic data member type
+  - a pointer or reference to a consteval-only type
+  - a function type with a consteval-only return type or a consteval-only parameter type
```

Comment:
I think if we copy the, uh ... structure ... from structural type (which has similar construction) you'd get...

A *consteval-only type* is one of the following:

* `std::meta::info`, or
* a pointer to, reference to, or (possibly multidimensional) array of a consteval-only type, or
* a function type whose return type or any parameter type is a consteval-only type, or
* a class type where the type of any base class or the type of any non-static data member is a consteval-only type. 

In the abstract they're equivalent, but I just think it's mildly better to have the symmetry.

Reply:
(I made a change along those lines, BTW.)